packages/angular/projects/clr-angular/src/data/datagrid/datagrid-row.ts

```diff
@@ -248,6 +247,15 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit
     this.subscriptions.forEach((sub: Subscription) => sub.unsubscribe());
   }
 
+  ngOnChanges(changes: any) {
+    /**
+     * When `clrDgSelectable` make sure  to lock or unlock the given row
+     */
+    if (changes.clrDgSelectable && changes.clrDgSelectable.currentValue !== changes.clrDgSelectable.previousValue) {
```

Comment:
The issue we are trying to solve was specifically in the **initial** input setting order.  And `ngOnInit` is what is there for us to coordinate the input properties because it runs once after all input changes.

So the fix could have been simply adding the following one line in the `ngOnInit` cycle instead of doing checks on every input value change:
```ts
this.selection.lockItem(this.item, !this.clrDgSelectable);
```